addons/account/wizard/account_payment_register.py

```diff
@@ -672,10 +672,11 @@ def _fetch_duplicate_reference(self, matching_states=('draft', 'posted')):
         """
         # Update tables involved in the query
         self.env['account.move.line'].flush_model(('move_id', 'payment_id', 'balance', 'account_id', 'company_id', 'date', 'partner_id'))
-        outstanding_account_ids = tuple(
-            (self.journal_id._get_journal_inbound_outstanding_payment_accounts() if self.payment_type == 'inbound'
-             else self.journal_id._get_journal_outbound_outstanding_payment_accounts()).ids
-        )
+
+        all_journals = self.env['account.journal'].search([])
```

Comment:
I think only bank and cash journal are relevant

Reply:
same applies everywhere
(only bank and cash journals should have payment method line ids, but it's just more accurate